src/dev/classes/OptionParser.ts

```diff
@@ -0,0 +1,187 @@
+interface ParsedOptions {
+    [key: string]: boolean | string;
+}
+
+interface Options {
+    [key: string]: "boolean" | "string";
+}
+
+/**
+ * OptionParser class provides functions for parsing command line arguments.
+ * 
+ * Functions for parsing command line arguments.
+ * 
+ * https://hexdocs.pm/elixir/1.13.4/OptionParser.html
+ * 
+ * ```ts
+    const argv: string[] = ["--debug", "--output", "output.txt"];
+    const opts: Options = { debug: "boolean", => "string" };
+    const parsedOptions = OptionParser.parse(argv, opts);
+    console.log(parsedOptions); // => { debug: true, => 'output.txt' }
+
+    const str = "--verbose --input input.txt";
+    const argvFromString = OptionParser.split(str);
+    console.log(argvFromString); // => [ '--verbose', '--input', 'input.txt' ]
+
+    const options = { verbose: true, input: "input.txt" };
+    const argvFromOptions = OptionParser.toArgv(options);
+    console.log(argvFromOptions); // => [ '--verbose', '--input', 'input.txt' ]
+```
+ */
+class OptionParser {
+    /**
+     * Parses the given `argv` array into a keyword list based on the provided `opts`.
+     * @param argv - The array of command line options and arguments.
+     * @param opts - The options configuration that specifies the type of each option.
+     * @returns The parsed options as a keyword list.
+     * @throws An Error if any invalid options are given.
+     */
+    static parse(argv: string[], opts: Options = {}): ParsedOptions {
+        const result: ParsedOptions = {};
+        let i = 0;
+
+        while (i < argv.length) {
+            const arg = argv[i];
+            if (!arg.startsWith("--")) {
+                break;
+            }
+
+            const option = arg.substring(2);
+            const nextArg = argv[i + 1];
+
+            switch (opts[option]) {
+                case "boolean":
+                    result[option] = true;
+                    i++;
+                    break;
+                case "string":
+                    if (nextArg) {
+                        result[option] = nextArg;
+                        i += 2;
+                    } else {
+                        throw new Error(`Invalid option: ${arg}`);
+                    }
+                    break;
+                default:
+                    throw new Error(`Invalid option: ${arg}`);
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Similar to `parse`, but only parses the head of `argv`.
+     * As soon as it finds a non-switch, it stops parsing.
+     * @param argv - The array of command line options and arguments.
+     * @param opts - The options configuration that specifies the type of each option.
+     * @returns The parsed options as a keyword list.
+     */
+    static parseHead(argv: string[], opts: Options = {}): ParsedOptions {
+        const result: ParsedOptions = {};
+        let i = 0;
```

Comment:
i is never used?

Reply:
it was, right below. but now moved it directly into the loop to be for(let i ...) lol